coriolis/osmorphing/base.py

```diff
@@ -393,39 +394,78 @@ def _set_selinux_autorelabel(self):
         except Exception as err:
             LOG.warning("Failed to set autorelabel: %r" % err)
 
-    def _configure_cloud_init_user_retention(self):
-        cloud_cfg_paths = ["/etc/cloud/cloud.cfg"]
+    def _write_cloud_init_mods_config(self, cloud_cfg):
+        cloud_config_path = "/etc/cloud/cloud.cfg.d/99_coriolis.cfg"
         cloud_cfgs_dir = "/etc/cloud/cloud.cfg.d"
-        if self._test_path(cloud_cfgs_dir):
-            for path in self._exec_cmd_chroot(
-                    'ls -1 %s' % cloud_cfgs_dir).decode().splitlines():
-                if path.endswith('.cfg'):
-                    cloud_cfg_paths.append("%s/%s" % (cloud_cfgs_dir, path))
-
+        if not self._test_path(cloud_cfgs_dir):
+            exception.CoriolisException("Could not find %s" % cloud_cfgs_dir)
```

Comment:
I think it'd be better to just create this directory instead of straight out erroring out. Cloud-init will look for this location regardless, it's not customizable afaik. (if it is, find a way to locate it)
At least execute: `mkdir -p /etc/cloud/cloud.cfg.d`